src/elements/earn/pools/addLiquidity/empty/AddLiquidityEmptyCTA.tsx

```diff
@@ -30,46 +31,43 @@ export const AddLiquidityEmptyCTA = ({
 }: Props) => {
   const dispatch = useDispatch();
   const { account } = useWeb3React();
+  const history = useHistory();
 
   const handleAddLiquidity = async () => {
-    try {
-      const data = [
-        { amount: amountBnt, token: bnt },
-        { amount: amountTkn, token: tkn },
-      ];
-      const txHash = await addLiquidity(data, pool.converter_dlt_id);
-      dispatch(
-        addNotification({
-          type: NotificationType.pending,
-          title: 'Add Liquidity',
-          msg: `You added ${prettifyNumber(amountTkn)} ${
-            tkn.symbol
-          } and ${prettifyNumber(amountBnt)} ${bnt.symbol} to pool ${
-            pool.name
-          }`,
+    const data = [
+      { amount: amountBnt, token: bnt },
+      { amount: amountTkn, token: tkn },
+    ];
+    const cleanTkn = prettifyNumber(amountTkn);
+    const cleanBnt = prettifyNumber(amountBnt);
+    await addLiquidity(
+      data,
+      pool.converter_dlt_id,
+      (txHash: string) =>
+        addLiquidityNotification(
+          dispatch,
           txHash,
-        })
-      );
-    } catch (e) {
-      console.error('Add liquidity failed with msg: ', e.message);
-      if (e.code === ErrorCode.DeniedTx) {
-        dispatch(
-          addNotification({
-            type: NotificationType.error,
-            title: 'Transaction Rejected',
-            msg: 'You rejected the transaction. If this was by mistake, please try again.',
-          })
-        );
-      } else {
-        dispatch(
-          addNotification({
-            type: NotificationType.error,
-            title: 'Transaction Failed',
-            msg: `Adding liquidity to pool ${pool.name} failed. Please try again or contact support.`,
-          })
-        );
-      }
-    }
+          cleanTkn,
+          tkn.symbol,
+          cleanBnt,
+          bnt.symbol,
+          pool.name
+        ),
+      () => {
+        if (window.location.pathname.includes(pool.pool_dlt_id))
+          history.push('/portfolio');
```

Comment:
We want to stop directly trigger navigation by components

Reply:
We can go over the code and create a route page with the strings and methods for things like checking url and pushing accordingly